lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -1436,8 +1440,36 @@ wantedErrorButSuccess =
     fail . ("expected an error but got a successful response: " <>) . show
 
 -- | Apply 'a' to all actions in sequence
-verify :: (Monad m) => a -> [a -> m ()] -> m ()
-verify a = mapM_ (a &)
+verify :: Show a => a -> [a -> IO ()] -> IO ()
+verify a = counterexample msg . mapM_ (a &)
+  where
+    msg = "While verifying " ++ show a
```

Comment:
I guess we could also show the index of the check… unless the `SrcLoc` already leads to it 🤔

Reply:
Looking at the screenshots above, it seems like the source location _now_ does point to the exact item within the `verify` list.